dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResource.java

```diff
@@ -0,0 +1,320 @@
+    private ServiceIntegrationHelper helper;
+
+    @VisibleForTesting
+    public ServiceIntegrationResource(final WebResource webResource,final ServiceIntegrationHelper helper) {
+        this.webResource = webResource;
+        this.helper = helper;
+    }
+
+    public ServiceIntegrationResource() {
+       this(new WebResource(),new ServiceIntegrationHelper());
+    }
+
+    @GET
+    @Path("/")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response listAvailableServices(@Context final HttpServletRequest request,
+                                                @Context final HttpServletResponse response
+    ) {
+        try {
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource)
+                            .requiredBackendUser(true)
+                            .requiredFrontendUser(false)
+                            .requestAndResponse(request, response)
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            final List<ServiceIntegrationView> integrationViews = helper.getAvailableDescriptorViews(user);
+            if(!integrationViews.isEmpty()){
+                return Response.ok(new ResponseEntityView(integrationViews)).build(); // 200
+            }
+            throw new DoesNotExistException("Nope. No service integration is available to configure. Try uploading a file!");
+        } catch (Exception e) {
+            Logger.error(this.getClass(), "Exception on listing available service integration descriptors.", e);
+            return ResponseUtil.mapExceptionResponse(e);
+        }
+    }
+
+    @GET
+    @Path("/{serviceKey}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getServiceIntegrationByKey(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @PathParam("serviceKey") final String serviceKey
+    ) {
+        try {
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource)
+                            .requiredBackendUser(true)
+                            .requiredFrontendUser(false)
+                            .requestAndResponse(request, response)
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            final Optional<ServiceIntegrationView> serviceIntegrationView = helper
+                    .getServiceIntegrationSiteView(serviceKey, user);
+            if (serviceIntegrationView.isPresent()) {
+                return Response.ok(new ResponseEntityView(serviceIntegrationView.get()))
+                        .build(); // 200
+            }
+            throw new DoesNotExistException(
```

Comment:
I don't now it the Resource should trow this exception mayba it should be thrown by the API or the helper